docs_user/modules/openstack-tls_everywhere.adoc

```diff
@@ -0,0 +1,211 @@
+[id="tls-everywhere_{context}"]
+
+//:context: tls
+
+//kgilliga: This module will be converted to an assembly. Check xref contexts.
+
+= TLS Everywhere
+
+_Disclaimer: the below steps were reproduced on a FreeIPA 4.10.1 server. The location of files and directories may slightly change on different versions._
+
+== Prerequisites
+
+* Check that the source deployment is using TLS Everywhere, skip this step if it's not.
+* Make sure the previous Adoption steps have been performed successfully.
+* Make sure the backend services on the new deployment are not started yet.
+
+== Variables
+
+Define the shell variables used in the following steps. The values are illustrative and refer to a single node standalone director deployment. Use values that are correct for your environment:
+
+----
+IPA_SSH="ssh -i ~/install_yamls/out/edpm/ansibleee-ssh-key-id_rsa root@<freeipa-server-ip-address>"
+----
+
+In this example the FreeIPA instance is running on a separate host, in a container.
+
+== Transfer the CA certificate and key from FreeIPA to the cert-manager Operator
+
+This is only going to work if:
```

Comment:
Replace with:

These instructions explain how to extract the CA signing certificate from the FreeIPA instance that is used to provide the certificates in the source environment and import it into certmanager for use in the target environment.  In this way, disruption on the compute nodes can be minimized because a new chain of trust need not be installed.

It is expected that the old FreeIPA node will then be decommissioned and no longer used to issue certificates,  This may not be possible if the IPA server is used to issue certificates for non-Openstack systems.

This procedure will also need to be modified if the signing keys are stored in an HSM (hardware security module) instead of an NSSDB.  In that case, if the key is retrievable, special HSM utilities may be required.